src/http/input/metadata/LinkRelParser.ts

```diff
@@ -13,20 +15,68 @@ import namedNode = DataFactory.namedNode;
 export class LinkRelParser extends MetadataParser {
   protected readonly logger = getLoggerFor(this);
 
-  private readonly linkRelMap: Record<string, NamedNode>;
+  private readonly linkRelMap: Record<string, LinkRelObject>;
 
-  public constructor(linkRelMap: Record<string, string>) {
+  public constructor(linkRelMap: Record<string, LinkRelObject>) {
     super();
-    this.linkRelMap = Object.fromEntries(
-      Object.entries(linkRelMap).map(([ header, uri ]): [string, NamedNode] => [ header, namedNode(uri) ]),
-    );
+    this.linkRelMap = linkRelMap;
   }
 
   public async handle(input: { request: HttpRequest; metadata: RepresentationMetadata }): Promise<void> {
     for (const { target, parameters } of parseLinkHeader(input.request.headers.link)) {
-      if (this.linkRelMap[parameters.rel]) {
-        input.metadata.add(this.linkRelMap[parameters.rel], namedNode(target));
+      this.linkRelMap[parameters.rel]?.addToMetadata(target, input.metadata, this.logger);
+    }
+  }
+}
+
+/**
+ * Represents the values that are parsed as metadata
+ */
+export class LinkRelObject {
+  public readonly value: NamedNode;
+  public readonly ephemeral: boolean;
+  public readonly filter: string[];
+
+  /**
+   * @param value - The value that will be used as predicate
+   * @param ephemeral - (Optional) Indicates whether it will be stored by the server
+   * @param filter - (Optional) Contains the only values that can be used with the given predicate
+   */
+  public constructor(value: string, ephemeral?: boolean, filter?: string[]) {
+    this.value = namedNode(value);
+    this.ephemeral = ephemeral ?? false;
+    this.filter = filter ?? [];
+  }
+
+  /**
+   * Calculates whether the object can be added to the metadata
+   * @param object - The link target
+   * @returns a boolean to indicate whether it can be added to the metadata or not
+   */
+  private objectAllowed(object: string): boolean {
+    return this.filter.length === 0 || this.filter.includes(object);
```

Comment:
If the array is empty, the new statement would return false instead of true

Reply:
So `filter` is an allow list, but an empty list means everything is allowed. This could lead to problems when you "add" an allowed item to the list, only to them disallow everything else.

How about we use `undefined` or `null` or `true` for that case where there is no allow list, and then `return this.allowList.?includes(object)`.

Reply:
I understand it can lead to problems, but we designed this class to be like that in order to either parse everything from the Link header (e.g. for `rel="http://www.w3.org/ns/ldp#inbox"` https://github.com/woutslabbinck/community-server/blob/feat/metadata-editing/config/ldp/metadata-parser/parsers/link.json#L24-L29 ) or to only allow certain values (e.g. for `https://github.com/woutslabbinck/community-server/blob/feat/metadata-editing/config/ldp/metadata-parser/parsers/link.json#L11-L21)

Reply:
That's okay; it was just a suggestion for the internal workings (just that there is a difference between passing in undefined as a third constructor argument, as opposed to an empty array, which would allow nothing).

Reply:
Would indeed differentiate between an empty list and an undefined list so people don't get confused (like I just was above).